### PR TITLE
Open license links in an overlay

### LINKED ISSUE #2550: Open license links in an overlay
I'm trying to develop this, but am having some trouble getting the Gateway to restart with a new webpacking.

Firstly, this is what I see after `npm install` on a Raspberry Pi 4 1Gb
```
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/eslint-plugin-html/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})
```

Running `npm start` shows this:

```
npm start

> webthings-gateway@0.12.0 start /home/pi/mozilla-iot/gateway
> webpack --display errors-only && node build/gateway.js

ℹ ｢atl｣: Using typescript@3.8.3 from typescript
ℹ ｢atl｣: Using tsconfig.json from /home/pi/mozilla-iot/gateway/tsconfig.json
ℹ ｢atl｣: Checking started in a separate process...
ℹ ｢atl｣: Time: 13213ms
```

Should it continue after this?

I am unable to open the gateway in the browser, even after waiting a few minutes.

### PR DESCRIPTION
**Pull request by flatsiedatsie**
Fixes #2550

### CODE REVIEW

**Comment by flatsiedatsie** [comment]
Is there a way to automatically fix 'lint' issues?

**Comment by flatsiedatsie** [comment]
I'm having trouble understanding the linter in some cases:

```
      API.getAddonsInfo().then((data) => {
        const licenseUrl =
          '${data.urls[0]}/license/${e.target.getAttribute("data-id")}';
```
Here the linter says that `'data' is defined but never used`, even though it is?

**Comment by flatsiedatsie** [comment]
Also, I'd like to lobby for increasing the maximum sentence length to something more 2020 :-)

**Comment by flatsiedatsie** [comment]
How can I lint CSS locally?

**Comment by flatsiedatsie** [comment]
There is still an error, but I don't believe it's related to my code

File: static/js/views/installed-addon.js [hunk 1]
```diff
@@ -249,6 +254,55 @@ class InstalledAddon {
...
       });
   }
+
+  /**
+   * Handle a click on the license button.
+   */
+  handleLicense(e) {
+    if (e.target.getAttribute('data-id')) {
+      API.getAddonsInfo().then((data) => {
+        const licenseUrl = `
```
Comment (mrstegeman) [inline static/js/views/installed-addon.js:264]: This is breaking the link. You're inserting space into it.

File: static/js/views/discovered-addon.js [hunk 2]
```diff
@@ -123,6 +128,55 @@ class DiscoveredAddon {
...
   }
+
+  /**
+   * Handle a click on the license button.
+   */
+  handleLicense(e) {
+    if (e.target.getAttribute('data-id')) {
+      API.getAddonsInfo().then((data) => {
+        const licenseUrl = `
+          ${data.urls[0]}/license/
```
Comment (mrstegeman) [inline static/js/views/discovered-addon.js:139]: This is breaking the link. You're inserting space into it.

File: static/js/views/installed-addon.js [hunk 3]
```diff
@@ -249,6 +254,55 @@ class InstalledAddon {
...
+          ${data.urls[0]}/license/
+          ${e.target.getAttribute('data-id')}`;
+        console.log(licenseUrl);
```
Comment (mrstegeman) [inline static/js/views/installed-addon.js:267]: Remove the log here.

File: static/js/views/installed-addon.js [hunk 4]
```diff
@@ -249,6 +254,55 @@ class InstalledAddon {
         button.disabled = false;
       });
   }
+
+  /**
+   * Handle a click on the license button.
+   */
+  handleLicense(e) {
+    if (e.target.getAttribute('data-id')) {
```
Comment (mrstegeman) [inline static/js/views/installed-addon.js:262]: Please use `e.target.dataset.id`

File: static/js/views/discovered-addon.js [hunk 5]
```diff
@@ -123,6 +128,55 @@ class DiscoveredAddon {
         controlDiv.innerHTML = el;
       });
   }
+
+  /**
+   * Handle a click on the license button.
+   */
+  handleLicense(e) {
+    if (e.target.getAttribute('data-id')) {
```
Comment (mrstegeman) [inline static/js/views/discovered-addon.js:136]: Please use `e.target.dataset.id`

File: static/js/views/discovered-addon.js [hunk 6]
```diff
@@ -123,6 +128,55 @@ class DiscoveredAddon {
...
+      API.getAddonsInfo().then((data) => {
+        const licenseUrl = `
+          ${data.urls[0]}/license/
+          ${e.target.getAttribute('data-id')}`;
+        console.log(licenseUrl);
```
Comment (mrstegeman) [inline static/js/views/discovered-addon.js:141]: Remove log.

File: static/js/views/installed-addon.js (hunk 1 above)
Comment (flatsiedatsie) [reply static/js/views/installed-addon.js:264]: If I remember correctly, if I don't immediately wrap like this, the linter give an error.

Comment (flatsiedatsie) [reply static/js/views/installed-addon.js]: Ah, it's the ` that was the issue.

File: static/js/views/discovered-addon.js [hunk 7]
```diff
@@ -123,6 +128,52 @@ class DiscoveredAddon {
...
+      API.getAddonsInfo().then((data) => {
+        const licenseUrl = `${data.urls[0]}/license/${e.target.dataset.id}`;
+        let modal = document.getElementById('media-modal');
+        if (modal == null) {
+          const modalContainer = document.createElement('div');
+          modalContainer.className = 'media-modal';
+          modalContainer.id = 'media-modal';
+          const modalFrame = document.createElement('div');
+          modalFrame.className = 'media-modal-frame';
+          const content = `'<div class="media-modal-close"
```
Comment (mrstegeman) [inline static/js/views/discovered-addon.js:146]: There is a stray `'` inside the template string, before the `<div>` tag.

File: static/js/views/installed-addon.js [hunk 8]
```diff
@@ -249,6 +254,52 @@ class InstalledAddon {
...
+      API.getAddonsInfo().then((data) => {
+        const licenseUrl = `${data.urls[0]}/license/${e.target.dataset.id}`;
+        let modal = document.getElementById('media-modal');
+        if (modal == null) {
+          const modalContainer = document.createElement('div');
+          modalContainer.className = 'media-modal';
+          modalContainer.id = 'media-modal';
+          const modalFrame = document.createElement('div');
+          modalFrame.className = 'media-modal-frame';
+          const content = `'<div class="media-modal-close"
```
Comment (mrstegeman) [inline static/js/views/installed-addon.js:272]: There is a stray `'` inside the template string, before the `<div>` tag.

File: static/css/addons-form.css [hunk 9]
```diff
@@ -219,3 +219,11 @@
 #addon-config-settings .button-submit:active {
   background-color: #658196;
 }
+
```
Comment (mrstegeman) [inline static/css/addons-form.css:222]: This should go in `settings.css`, along with the other styles for this element. Also, the link should always be white. To make that happen, change this block in `settings.css`:

```css
.addon-settings-license,
.addon-settings-author {
  font-size: 1.4rem;
  font-style: italic;
  color: #ccc;
}

.addon-settings-license {
  padding-left: 0.5rem;
}

.addon-settings-license a:link,
.addon-settings-license a:visited,
.addon-settings-license a:hover,
.addon-settings-license a:active,
.addon-settings-author a:link,
.addon-settings-author a:visited,
.addon-settings-author a:hover,
.addon-settings-author a:active {
  color: #fff;
}
```

to:

```css
.addon-settings-license,
.addon-settings-author {
  font-size: 1.4rem;
  font-style: italic;
}

.addon-settings-author {
  color: #ccc;
}

.addon-settings-author a:link,
.addon-settings-author a:visited,
.addon-settings-author a:hover,
.addon-settings-author a:active {
  color: #fff;
}

.addon-settings-license {
  padding-left: 0.5rem;
  cursor: pointer;
  color: #fff;
}
```

File: static/js/views/discovered-addon.js (hunk 7 above)
Comment (flatsiedatsie) [reply static/js/views/discovered-addon.js:146]: well spotted

File: static/css/addons-form.css (hunk 9 above)
Comment (flatsiedatsie) [reply static/css/addons-form.css:222]: done

**Comment by flatsiedatsie** [comment]
Would you be willing to change the `settings.css` file?

**Pull request merged**